activerecord/lib/active_record/railties/databases.rake

```diff
@@ -295,24 +295,23 @@ db_namespace = namespace :db do
     seed = false
 
     ActiveRecord::Base.configurations.configs_for(env_name: ActiveRecord::Tasks::DatabaseTasks.env).each do |db_config|
-      ActiveRecord::Base.establish_connection(db_config.config)
-
-      ActiveRecord::Tasks::DatabaseTasks.migrate
-
-      # Skipped when no database
-      ActiveRecord::Tasks::DatabaseTasks.dump_schema(db_config.config, ActiveRecord::Base.schema_format, db_config.spec_name)
-
-    rescue ActiveRecord::NoDatabaseError
-      ActiveRecord::Tasks::DatabaseTasks.create_current(db_config.env_name, db_config.spec_name)
-      ActiveRecord::Tasks::DatabaseTasks.load_schema(
-        db_config.config,
-        ActiveRecord::Base.schema_format,
-        nil,
-        db_config.env_name,
-        db_config.spec_name
-      )
+      if ActiveRecord::Base.database_exists?(db_config.config)
+        ActiveRecord::Tasks::DatabaseTasks.migrate
+        ActiveRecord::Tasks::DatabaseTasks.dump_schema(db_config.config, ActiveRecord::Base.schema_format, db_config.spec_name)
+      else
+        ActiveRecord::Tasks::DatabaseTasks.create_current(db_config.env_name, db_config.spec_name)
+        ActiveRecord::Tasks::DatabaseTasks.dump_schema(db_config.config, ActiveRecord::Base.schema_format, db_config.spec_name)
```

Comment:
@robertomiranda Why did you add dumping schema in this place? There is actually nothing to dump as the database is created step above